hugr/src/algorithm/verify.rs

```diff
@@ -0,0 +1,102 @@
+//! Provides [VerifyLevel] with tools to run passes with configurable
+//! verification.
+
+use thiserror::Error;
+
+use crate::{
+    extension::ExtensionRegistry,
+    hugr::{HugrMut, ValidationError},
+    HugrView,
+};
+
+#[derive(Debug, Clone, Copy, Ord, Eq, PartialOrd, PartialEq)]
+/// A type for running [HugrMut] algorithms with verification.
+///
+/// Provides [VerifyLevel::run_verified_pass] to invoke a closure with pre and
+/// post verification.
+///
+/// The default level is `None` because verification can be expensive.
+pub enum VerifyLevel {
```

Comment:
is there a clear distinction between "validation" and "verification"? Should this be called "ValidateLevel" for consistency?

Reply:
That makes sense, though I worry people would expect some kind of formal verification on passes. I think for something similar in tket1 we used the term "audit" though I don't like that much either.